src/ui/Analytics/Analytics.ts

```diff
@@ -158,7 +162,69 @@ export class Analytics extends Component {
      * Default value is `undefined`, and the value of this parameter will fallback to the organization used for the
      * search endpoint.
      */
-    organization: ComponentOptions.buildStringOption()
+    organization: ComponentOptions.buildStringOption(),
+
+    /**
+     * Whether to push Coveo usage analytics events to the Google Tag Manager [data layer](https://developers.google.com/tag-manager/devguide#datalayer).
+     *
+     * **Notes:**
+     * - Set this option to `false` to ensure that the component does _not_ populate the data layer, even if a GTM container snippet is present in the page.
+     * - Set this option to `true` to ensure that the component populates the data layer even if a GTM container snippet cannot be found in the page.
+     *
+     * See also [`gtmDataLayerName`]{@link Analytics.options.gtmDataLayerName}.
+     *
+     * By default, if a standard GTM container snippet is found in the page, the component automatically attempts to push to its initialized data layer (see [Google Tag Manager Quick Start Guide](https://developers.google.com/tag-manager/quickstart)).
+     */
+    pushToGtmDataLayer: ComponentOptions.buildBooleanOption({
+      postProcessing: (value?: boolean, options?: IAnalyticsOptions) => {
+        const logger = new Logger('Analytics');
+        let optionValue = false;
+        if (typeof value === 'boolean') {
+          optionValue = value;
+        } else {
+          const includesGtmScript = new RegExp(`https:\/\/www\.googletagmanager\.com\/gtm\.js`);
+          const initializesGtmScript = new RegExp(`\(window,document,'script','(\\w+)','GTM-[A-Z0-9]{7}'\)`);
+          const gtmContainerSnippet = _.find(document.querySelectorAll('script'), scriptElement => {
+            // Could yield false positives (e.g., if the innerText is commented out).
+            return includesGtmScript.test(scriptElement.innerText) && initializesGtmScript.test(scriptElement.innerText);
+          });
+          if (gtmContainerSnippet) {
+            const match = initializesGtmScript.exec(gtmContainerSnippet.innerText);
+            // Attempt to extract data layer name from GTM container snippet
+            if (match.length > 2) {
+              options.gtmDataLayerName = match[2];
```

Comment:
I would have liked to use a named group in the regular expression and refer to this, rather than using the index position to access the string I need in the match array. However, I can't seem to find a way to do this. Any ideas?

Reply:
If you just want the original string, would it be possible to set:
`options.gtmDataLayerName = gtmContainerSnippet.innerText`

Or does `match[2]` hold a different value?